core/object/class_db.h

```diff
@@ -171,7 +171,7 @@ class ClassDB {
 
 	template <typename T>
 	static Object *creator(bool p_notify_postinitialize) {
-		Object *ret = new ("") T;
+		Object *ret = new (DefaultAllocator{}) T;
```

Comment:
```suggestion
		Object *ret = memnew(T);
```

Reply:
Ha, I fell into this trap as well. `memnew` is not possible here, because it calls the objects' postinitialize, which we need to have control over (via `p_notify_postinitialize`) in this function.
Took me quite a while to debug in an earlier version.

Reply:
Oh damn, that's subtle! I'd recommend adding a comment with an explanation, as that would've stumped me for a good while too

Reply:
Good idea!